robot_sf/nav/motion_planning_adapter.py

```
@@ -0,0 +1,365 @@
+    def _set_world_axis_formatters(self, grid_map: Grid, meters_per_cell: float | None) -> None:
+        scale_factor = self._resolve_meters_per_cell(grid_map, meters_per_cell)
+        if scale_factor is None or grid_map.dim != 2:
+            return
+
+        formatter = FuncFormatter(lambda value, _: f"{value * scale_factor:.2f}")
+        self.ax.xaxis.set_major_formatter(formatter)
+        self.ax.yaxis.set_major_formatter(formatter)
+        self.ax.set_xlabel("X (m)")
+        self.ax.set_ylabel("Y (m)")
+
+    def plot_grid_map(  # type: ignore[override]
```

Pylint (4.0.4)

[refactor] Too many arguments (7/5)

(R0913)